### PR TITLE
[capture] Add capture script for OTBN vertical analysis

### PR DESCRIPTION
**Pull request by wettermo**
This PR

- addresses issue https://github.com/lowRISC/ot-sca/issues/216
- adds a capture script for OTBN vertical analysis, which supports "keygen" and "modinv" OTBN apps.
      -> For this, capture_aes.py is used as a template.
- updates the OTBN vertical device binary.

### CODE REVIEW

File: capture/capture_otbn.py [hunk 1]
```diff
@@ -0,0 +1,764 @@
...
+from scopes.cycle_converter import convert_num_cycles, convert_offset_cycles
+from scopes.scope import Scope, ScopeConfig, determine_sampling_rate
+from tqdm import tqdm
+
+sys.path.append("../")
+import util.helpers as helpers  # noqa: E402
+from target.cw_fpga import CWFPGA  # noqa: E402
+from util import check_version  # noqa: E402
+from util import plot  # noqa: E402
+from util import data_generator as dg  # noqa: E402
```
Comment (nasahlpa) [inline capture/capture_otbn.py:29]:
```suggestion
import util.helpers as helpers
from target.cw_fpga import CWFPGA
from util import check_version
from util import data_generator as dg
from util import plot
```
Luckily we don't need the `sys.path.append("../")` anymore.

File: capture/capture_otbn.py [hunk 2]
```diff
@@ -0,0 +1,764 @@
...
+    # `adc_mul` affects the sample frequency (clock_freq = adc_mul * pll_freq)
+    # `decimate` is the ADC downsampling factor that allows us to sample at
+    #  every `decimate` cycles.
+    if scope_type == "husky":
+        if "adc_mul" in cfg["husky"]:
+            scope.scope.scope.clock.adc_mul = cfg["husky"]["adc_mul"]
+        if "decimate" in cfg["husky"]:
+            scope.scope.scope.adc.decimate = cfg["husky"]["decimate"]
+        # Print final scope parameter
+        print(
```
Comment (nasahlpa) [inline capture/capture_otbn.py:157]:
```suggestion
        logger.info(
```
I suggest to stick to `logger.info` inside the script.

File: capture/capture_otbn.py [hunk 3]
```diff
@@ -0,0 +1,764 @@
...
+        config: The capture configuration.
+    """
+    if config["capture"]["show_plot"]:
+        plot.save_plot_to_file(
+            project.get_waves(0, config["capture"]["plot_traces"]),
+            set_indices=None,
+            num_traces=config["capture"]["plot_traces"],
+            outfile=config["capture"]["trace_image_filename"],
+            add_mean_stddev=True)
+        print(f'Created plot with {config["capture"]["plot_traces"]} traces: '
```
Comment (nasahlpa) [inline capture/capture_otbn.py:671]:
```suggestion
        logger.info(f'Created plot with {config["capture"]["plot_traces"]} traces: '
```

File: capture/capture_otbn.py [hunk 4]
```diff
@@ -0,0 +1,764 @@
...
+    metadata["offset_samples"] = scope.scope_cfg.offset_samples
+    metadata["scope_gain"] = scope.scope_cfg.scope_gain
+    metadata["cfg_file"] = str(args.cfg)
+    metadata["fpga_bitstream"] = cfg["target"]["fpga_bitstream"]
+    # TODO: Store binary into database instead of binary path.
+    # (Issue lowrisc/ot-sca#214)
+    metadata["fw_bin"] = cfg["target"]["fw_bin"]
+    # TODO: Allow user to enter notes via CLI.
+    # (Issue lowrisc/ot-sca#213)
+    metadata["notes"] = ""
```
Comment (nasahlpa) [inline capture/capture_otbn.py:756]:
```suggestion
    # Store bitstream information.
    metadata["fpga_bitstream_path"] = cfg["target"]["fpga_bitstream"]
    metadata["fpga_bitstream_crc"] = helpers.file_crc(cfg["target"]["fpga_bitstream"])
    if args.save_bitstream:
        metadata["fpga_bitstream"] = helpers.get_binary_blob(cfg["target"]["fpga_bitstream"])
    # Store binary information.
    metadata["fw_bin_path"] = cfg["target"]["fw_bin"]
    metadata["fw_bin_crc"] = helpers.file_crc(cfg["target"]["fw_bin"])
    if args.save_binary:
        metadata["fw_bin"] = helpers.get_binary_blob(cfg["target"]["fw_bin"])
    # Store user provided notes.
    metadata["notes"] = args.notes
    # Store the Git hash.
    metadata["git_hash"] = helpers.get_git_hash()
```
Could you please add the following to reflect recent changes to the metadata?

**Review by nasahlpa — APPROVED** [review]
Thanks for working on this @wettermo!

I've added some small comments.

File: capture/capture_otbn.py (hunk 4 above)
Comment (wettermo) [reply capture/capture_otbn.py]: Of course! Sorry, I apparently used an older version of capture_aes.py as a template and missed the changes when rebasing.

**Comment by wettermo** [comment]
Thanks Pascal for the review! I added all your suggestions to the code.
In case you don't have any objections anymore, I would merge the PR.

**Pull request merged**